modules/ui/maproulette_details.js

```diff
@@ -1,79 +1,131 @@
+import { select as d3_select } from 'd3-selection';
+
+import { utilHighlightEntities } from '../util/index.js';
+
 export function uiMapRouletteDetails(context) {
   const l10n = context.systems.l10n;
   const maproulette = context.services.maproulette;
 
   let _qaItem;
 
 
+  function parseShortCodes(text) {
+    const segments = text.split(/\[select\s+&quot;\s*[^"]*?\s*&quot;\s+name=&quot;/);
+    let transformedText = segments[0];
+    segments.slice(1).forEach(segment => {
+      const endIndex = segment.indexOf('&quot;');
+      const dropdownName = segment.substring(0, endIndex);
+      const valuesStart = segment.indexOf('values=&quot;') + 'values=&quot;'.length;
+      const valuesEnd = segment.indexOf('&quot;', valuesStart);
+      const options = segment.substring(valuesStart, valuesEnd).split(',');
+      const dropdownHtml = `<select name="${dropdownName}"><option value=""></option>${options.map(option => `<option value="${option.trim()}">${option.trim()}</option>`).join('')}</select>`;
+      const remainder = segment.substring(valuesEnd + '&quot;'.length).trim().replace(/^\]/, ''); // Remove the first closing bracket if it exists
+      transformedText += dropdownHtml + remainder;
```

Comment:
This method is a bit hard to read/parse through to understand without knowing the content flowing in. Could it be made a bit more modular? perhaps a 'generateMRDropdown' method to cleanly call out inputs/outputs that are used to generate the 'dropdownHtml' on line 21.

Reply:
Oh haha I guess this is the same thing I commented on, just moved to a new function.